### PR TITLE
benchmark: Add TPC-H benchmark for v4.0 vs v3.0

### PR DESCRIPTION
**Pull request by JaySon-Huang**
Signed-off-by: JaySon-Huang <tshent@qq.com>

### What is changed, added or deleted? (Required)

Add TPC-H benchmark for v4.0 vs v3.0

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

File: benchmark/v4.0-performance-benchmarking-with-tpch.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
+---
+title: TiDB TPC-H 性能对比测试报告 - v4.0 对比 v3.0
+category: benchmark
+draft: true
```
Comment (zz-jason) [inline benchmark/v4.0-performance-benchmarking-with-tpch.md:4]:
```suggestion
draft: false
```

should it be false?

File: benchmark/v4.0-performance-benchmarking-with-tpch.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+#### session 变量
+
+- set global tidb_distsql_scan_concurrency = 30;
+- set global tidb_distsql_scan_concurrency = 30;
+- set global tidb_projection_concurrency = 16;
+- set global tidb_hashagg_partial_concurrency = 16;
+- set global tidb_hashagg_final_concurrency = 16;
+- set global tidb_hash_join_concurrency = 16;
+- set global tidb_index_lookup_concurrency = 16;
+- set global tidb_index_lookup_join_concurrency = 16;
```
Comment (zz-jason) [inline benchmark/v4.0-performance-benchmarking-with-tpch.md:49]: How about this:

{{< copyable "sql" >}}

```sql
set global tidb_distsql_scan_concurrency = 30;
set global tidb_distsql_scan_concurrency = 30;
set global tidb_projection_concurrency = 16;
set global tidb_hashagg_partial_concurrency = 16;
set global tidb_hashagg_final_concurrency = 16;
set global tidb_hash_join_concurrency = 16;
set global tidb_index_lookup_concurrency = 16;
set global tidb_index_lookup_join_concurrency = 16;
```

File: benchmark/v4.0-performance-benchmarking-with-tpch.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+- set global tidb_allow_batch_cop = 1;
+- set global tidb_opt_distinct_agg_push_down = 1;
+- set global tidb_distsql_scan_concurrency = 30;
+- set global tidb_projection_concurrency = 16;
+- set global tidb_hashagg_partial_concurrency = 16;
+- set global tidb_hashagg_final_concurrency = 16;
+- set global tidb_hash_join_concurrency = 16;
+- set global tidb_index_lookup_concurrency = 16;
+- set global tidb_index_lookup_join_concurrency = 16;
```
Comment (zz-jason) [inline benchmark/v4.0-performance-benchmarking-with-tpch.md:79]: ditto

File: benchmark/v4.0-performance-benchmarking-with-tpch.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+- 硬件准备。
+为了避免 TiKV 及 TiFlash 磁盘 IO 资源上的争抢，把 EC2 带的两个 NVMe SSD 分别挂载为 `/data1` 及 `/data2`，把 TiKV 的部署至 `/data1`，而 TiFlash 部署至 `/data2`。
+
+- 通过 TiUP 部署 TiDB v4.0 和 v3.0。
+
+- 通过 TiUP 的 bench 工具导入 TPC-H 10 数据。
+  * v3.0 导入数据命令：
+
+    ```bash
```
Comment (zz-jason) [inline benchmark/v4.0-performance-benchmarking-with-tpch.md]: It's better to add:

{{< copyable "bash" >}}

File: benchmark/v4.0-performance-benchmarking-with-tpch.md [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+    tiup bench tpch prepare \
+    --host ${tidb_v3_host} --port ${tidb_v3_port} --db tpch_10 \
+    --sf 10 \
+    --analyze --tidb_build_stats_concurrency 8 --tidb_distsql_scan_concurrency 30
+    ```
+
+  * v4.0 导入数据命令：
+
+    ```bash
```
Comment (zz-jason) [inline benchmark/v4.0-performance-benchmarking-with-tpch.md]: ditto

**Comment by shuke987** [comment]
LGTM

File: benchmark/v4.0-performance-benchmarking-with-tpch.md [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+
+- 运行 TPC-H 的查询。
+
+  * 下载 TPC-H 的 sql 查询文件 `git clone https://github.com/pingcap/tidb-bench.git && cd tpch/queries`。
+
+  * 对于 TiDB v3.0，使用 mysql 客户端连接到 TiDB，然后执行查询，记录 v3.0 查询耗时。
+
+  * 对于 TiDB v4.0，使用 mysql 客户端连接到 TiDB
+
+    * 先设置 `set @@session.tidb_isolation_read_engines = 'tikv';` 后，再执行查询，记录 v4.0 仅读取 TiKV 的查询耗时。
```
Comment (zz-jason) [inline benchmark/v4.0-performance-benchmarking-with-tpch.md]:
```suggestion
    * 先设置 `set @@session.tidb_isolation_read_engines = 'tikv,tidb';` 后，再执行查询，记录 v4.0 仅读取 TiKV 的查询耗时。
```

File: benchmark/v4.0-performance-benchmarking-with-tpch.md [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+
+    * 先设置 `set @@session.tidb_isolation_read_engines = 'tikv,tiflash';` 后，再执行查询，记录 v4.0 通过智能选择混合读取 TiKV、TiFlash 的查询耗时。
```
Comment (zz-jason) [inline benchmark/v4.0-performance-benchmarking-with-tpch.md]:
```suggestion
    * 先设置 `set @@session.tidb_isolation_read_engines = 'tikv,tiflash,tidb';` 后，再执行查询，记录 v4.0 通过智能选择混合读取 TiKV、TiFlash 的查询耗时。
```

File: benchmark/v4.0-performance-benchmarking-with-tpch.md (hunk 1 above)
Comment (JaySon-Huang) [reply benchmark/v4.0-performance-benchmarking-with-tpch.md:4]: Now tpcc benchresult is true, I just keep that for consistent

File: benchmark/v4.0-performance-benchmarking-with-tpch.md [hunk 8]
```diff
@@ -0,0 +1,167 @@
+---
+title: TiDB TPC-H 性能对比测试报告 - v4.0 对比 v3.0
+category: benchmark
+draft: true
+---
+
+# TiDB TPC-H 性能对比测试报告 - v4.0 对比 v3.0
+
```
Comment (TomShawn) [inline benchmark/v4.0-performance-benchmarking-with-tpch.md:8]: 本文档对比 TiDB v4.0 和 v3.0 的性能测试结果。

File: benchmark/v4.0-performance-benchmarking-with-tpch.md [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+## 测试目的
+
+对比 TiDB v4.0 和 v3.0 OLAP 场景下的性能。
+
+因为 TiDB v4.0 中新引入了 [TiFlash](/tiflash/tiflash-overview.md) 组件增强 TiDB HTAP 形态。 本文会比较 v3.0 与 v4.0 仅读取 TiKV， v4.0 通过智能选择混合读取 TiKV、TiFlash 三者的性能。
```
Comment (TomShawn) [inline benchmark/v4.0-performance-benchmarking-with-tpch.md:13]: 三者吗？这里只写了 TiKV 和 TiFlash 两者。
```suggestion
因为 TiDB v4.0 中新引入了 [TiFlash](/tiflash/tiflash-overview.md) 组件增强 TiDB HTAP 形态，本文的测试对象如下：

+ v3.0 仅从 TiKV 读取数据。
+ v4.0 仅从 TiKV 读取数据。
+ v4.0 通过智能选择混合读取 TiKV、TiFlash 的数据。
```

File: benchmark/v4.0-performance-benchmarking-with-tpch.md [hunk 10]
```diff
@@ -0,0 +1,167 @@
...
+| PD              | m5.xlarge   |  3  |
+| TiDB            | c5.4xlarge  |  2  |
+| TiKV & TiFlash  | i3.4xlarge  |  3  |
+| TPC-H           | m5.xlarge   |  1  |
+
+### 软件版本
+
+| 服务类型   | 软件版本        
+|:----------|:-----------|
+| PD        | 3.0、4.0   |
```
Comment (TomShawn) [inline benchmark/v4.0-performance-benchmarking-with-tpch.md:30]: 需要写是 4.0 和 3.0 的具体哪个版本吗？

File: benchmark/v4.0-performance-benchmarking-with-tpch.md [hunk 11]
```diff
@@ -0,0 +1,167 @@
...
+| TiDB      | 3.0、4.0   |
+| TiKV      | 3.0、4.0   |
+| TiFlash   | 4.0        |
+| tiup-bench | 0.2      |
+
+### 配置参数
+
+#### v3.0
+
+##### 全局变量
```
Comment (TomShawn) [inline benchmark/v4.0-performance-benchmarking-with-tpch.md:40]:
```suggestion
**v3.0 的全局变量配置如下：**
```

File: benchmark/v4.0-performance-benchmarking-with-tpch.md [hunk 12]
```diff
@@ -0,0 +1,167 @@
...
+set global tidb_hashagg_partial_concurrency = 16;
+set global tidb_hashagg_final_concurrency = 16;
+set global tidb_hash_join_concurrency = 16;
+set global tidb_index_lookup_concurrency = 16;
+set global tidb_index_lookup_join_concurrency = 16;
+```
+
+#### v4.0
+
+##### TiKV
```
Comment (TomShawn) [inline benchmark/v4.0-performance-benchmarking-with-tpch.md:56]:
```suggestion
**v4.0 的 TiKV 配置如下**
```

File: benchmark/v4.0-performance-benchmarking-with-tpch.md [hunk 13]
```diff
@@ -0,0 +1,167 @@
...
+
+readpool.storage.use-unified-pool: false
+
+readpool.coprocessor.use-unified-pool: true
+
+##### PD
```
Comment (TomShawn) [inline benchmark/v4.0-performance-benchmarking-with-tpch.md:62]:
```suggestion
**v4.0 的 PD 配置如下**
```

File: benchmark/v4.0-performance-benchmarking-with-tpch.md [hunk 14]
```diff
@@ -0,0 +1,167 @@
...
+set global tidb_index_lookup_join_concurrency = 16;
+```
+
+#### v4.0
+
+##### TiKV
+
+readpool.storage.use-unified-pool: false
+
+readpool.coprocessor.use-unified-pool: true
```
Comment (TomShawn) [inline benchmark/v4.0-performance-benchmarking-with-tpch.md:60]: 请写进代码块里，并在 ``` 处标明是什么语言。

File: benchmark/v4.0-performance-benchmarking-with-tpch.md [hunk 15]
```diff
@@ -0,0 +1,167 @@
...
+
+##### PD
+
+replication.enable-placement-rules: true
+
+##### TiFlash
+
+logger.level: "info"
+
+learner_config.log-level: "info"
```
Comment (TomShawn) [inline benchmark/v4.0-performance-benchmarking-with-tpch.md:70]: ditto

File: benchmark/v4.0-performance-benchmarking-with-tpch.md [hunk 16]
```diff
@@ -0,0 +1,167 @@
...
+
+readpool.storage.use-unified-pool: false
+
+readpool.coprocessor.use-unified-pool: true
+
+##### PD
+
+replication.enable-placement-rules: true
+
+##### TiFlash
```
Comment (TomShawn) [inline benchmark/v4.0-performance-benchmarking-with-tpch.md:66]:
```suggestion
**v4.0 的 TiFlash 配置如下**
```

File: benchmark/v4.0-performance-benchmarking-with-tpch.md [hunk 17]
```diff
@@ -0,0 +1,167 @@
...
+
+logger.level: "info"
+
+learner_config.log-level: "info"
+
+##### 全局变量
```
Comment (TomShawn) [inline benchmark/v4.0-performance-benchmarking-with-tpch.md:72]:
```suggestion
**v4.0 的全局变量配置如下**
```

File: benchmark/v4.0-performance-benchmarking-with-tpch.md (hunk 1 above)
Comment (zz-jason) [reply benchmark/v4.0-performance-benchmarking-with-tpch.md:4]: @shuke987 @lilin90 please help us to confirm whether it should be `false`.

File: benchmark/v4.0-performance-benchmarking-with-tpch.md (hunk 10 above)
Comment (JaySon-Huang) [reply benchmark/v4.0-performance-benchmarking-with-tpch.md:30]: 舒科老师说只提 3.0 vs 4.0

Comment (TomShawn) [reply benchmark/v4.0-performance-benchmarking-with-tpch.md:30]: 好的

File: benchmark/v4.0-performance-benchmarking-with-tpch.md (hunk 8 above)
Comment (JaySon-Huang) [reply benchmark/v4.0-performance-benchmarking-with-tpch.md:8]: 不需要保留一级标题吗？
https://github.com/pingcap/docs-cn/pull/3166/files/471562b5f543611fbaf94d0773ef16a8acfbc9fe#r430512415

File: benchmark/v4.0-performance-benchmarking-with-tpch.md (hunk 14 above)
Comment (JaySon-Huang) [reply benchmark/v4.0-performance-benchmarking-with-tpch.md:64]: 这部分是部署时配置文件(yaml)中的一些选项。我看 tpc-c 的也是这样风格。需要放进代码块里吗？

File: benchmark/v4.0-performance-benchmarking-with-tpch.md (hunk 8 above)
Comment (TomShawn) [reply benchmark/v4.0-performance-benchmarking-with-tpch.md:8]: 要保留一级标题，推荐一二级标题间加一句简介。

Comment (JaySon-Huang) [reply benchmark/v4.0-performance-benchmarking-with-tpch.md:8]: 如果需要内容摘要的话，紧跟着的测试目的其实就是摘要。。

File: benchmark/v4.0-performance-benchmarking-with-tpch.md (hunk 14 above)
Comment (TomShawn) [reply benchmark/v4.0-performance-benchmarking-with-tpch.md:64]: 需要，请将两篇文档相应部分一起放进去 ```yaml 块中，谢啦

File: benchmark/v4.0-performance-benchmarking-with-tpch.md (hunk 8 above)
Comment (TomShawn) [reply benchmark/v4.0-performance-benchmarking-with-tpch.md:8]: 好的

File: benchmark/v4.0-performance-benchmarking-with-tpch.md (hunk 14 above)
Comment (JaySon-Huang) [reply benchmark/v4.0-performance-benchmarking-with-tpch.md]: Addressed.

File: benchmark/v4.0-performance-benchmarking-with-tpch.md [hunk 18]
```diff
@@ -0,0 +1,181 @@
...
+        --analyze --tidb_build_stats_concurrency 8 --tidb_distsql_scan_concurrency 30
+      ```
+
+- 运行 TPC-H 的查询。
+
+    * 下载 TPC-H 的 sql 查询文件 `git clone https://github.com/pingcap/tidb-bench.git && cd tpch/queries`。
+
+    * 对于 TiDB v3.0，使用 mysql 客户端连接到 TiDB，然后执行查询，记录 v3.0 查询耗时。
+
+    * 对于 TiDB v4.0，使用 mysql 客户端连接到 TiDB
```
Comment (JaySon-Huang) [inline benchmark/v4.0-performance-benchmarking-with-tpch.md:140]:
```suggestion
    * 对于 TiDB v4.0，使用 mysql 客户端连接到 TiDB，再根据需要测试的形态，选择以下其中一种操作：
```

**Comment by JaySon-Huang** [comment]
> @JaySon-Huang tpcc 没有测试结果，请添加

tpcc 的结果需要由 @innerr @Connor1996  更新

File: TOC.md [hunk 19]
```diff
@@ -36,6 +36,8 @@
   + 性能测试报告及重现指南
     + [如何用 Sysbench 测试 TiDB](/benchmark/benchmark-tidb-using-sysbench.md)
     + [如何对 TiDB 进行 TPC-C 测试](/benchmark/benchmark-tidb-using-tpcc.md)
+    + [TPC-C 性能对比 - v4.0 对比 v3.0](/benchmark/v4.0-performance-benchmarking-with-tpcc.md)
```
Comment (TomShawn) [inline TOC.md:39]:
```suggestion
```

File: benchmark/v4.0-performance-benchmarking-with-tpcc.md [hunk 20]
```diff
@@ -14,7 +14,7 @@ draft: true
 
 ### 硬件配置
 
-| 服务类型   | EC2 类型   |    节点数  |      
+| 服务类型   | EC2 类型   |    节点数  |
```
Comment (CaitinChen) [inline benchmark/v4.0-performance-benchmarking-with-tpcc.md:17]: 叫“节点数”而不是“实例数”？

File: benchmark/v4.0-performance-benchmarking-with-tpch.md (hunk 1 above)
Comment (lilin90) [reply benchmark/v4.0-performance-benchmarking-with-tpch.md:4]: If the testing results in this doc are ready to publish, then we can remove this `draft` line. Just keep it when the content is not ready.

File: benchmark/v4.0-performance-benchmarking-with-tpch.md [hunk 21]
```diff
@@ -0,0 +1,188 @@
+---
+title: TiDB TPC-H 性能对比测试报告 - v4.0 对比 v3.0
+category: benchmark
+draft: true
```
Comment (lilin90) [inline benchmark/v4.0-performance-benchmarking-with-tpch.md:4]:
```suggestion
```

**Comment by JaySon-Huang** [comment]
Should we wait for TPC-C / Sysbench benchmark result ready before merging this PR? @shuke987 @zz-jason

**Comment by JaySon-Huang** [comment]
Should we wait for TPC-C / Sysbench benchmark result ready before merging this PR? @shuke987 @zz-jason

File: benchmark/v4.0-performance-benchmarking-with-tpcc.md (hunk 20 above)
Comment (TomShawn) [reply benchmark/v4.0-performance-benchmarking-with-tpcc.md]: @CaitinChen PTAL

File: benchmark/v4.0-performance-benchmarking-with-tpch.md (hunk 1 above)
Comment (JaySon-Huang) [reply benchmark/v4.0-performance-benchmarking-with-tpch.md]: Removed.

File: benchmark/v4.0-performance-benchmarking-with-tpcc.md [hunk 22]
```diff
@@ -14,7 +14,7 @@ draft: true
```
Comment (yikeke) [inline benchmark/v4.0-performance-benchmarking-with-tpcc.md:14]: The `draft: true` in meta info need to be deleted if. you want this file public on website. @JaySon-Huang

Comment (JaySon-Huang) [reply benchmark/v4.0-performance-benchmarking-with-tpcc.md:14]: I have removed that for TPC-H benchmark result. The TPC-C one should be confirmed by @innerr @Connor1996 .

Comment (JaySon-Huang) [reply benchmark/v4.0-performance-benchmarking-with-tpcc.md:14]: And not adding TPC-H result to `TOC.md` until @shuke987 @zz-jason confirm that we can show TPC-H result on our website.

Comment (yikeke) [reply benchmark/v4.0-performance-benchmarking-with-tpcc.md:14]: Got it.

**Comment by JaySon-Huang** [comment]
Merge this PR with confirmed TPC-H benchmark result. But not showing it in TOC.md by now. 

Maybe show it with confirmed TPC-C, Sysbench result later.

**Pull request merged**